l1-contracts/src/UniFiAVSManager.sol

```diff
@@ -577,5 +782,70 @@ contract UniFiAVSManager is UniFiAVSManagerStorage, IUniFiAVSManager, UUPSUpgrad
         emit DeregistrationDelaySet(oldDelay, newDelay);
     }
 
+    function _validateProofAndGetValidatorKeys(BeaconChainHelperLib.InclusionProof memory proof, bool checkIndex)
+        internal
+        returns (uint64, bytes32)
+    {
+        UniFiAVSStorage storage $ = _getUniFiAVSManagerStorage();
+
+        if (!BeaconChainHelperLib.verifyValidatorExistence(proof)) {
+            revert InvalidValidatorProof(proof.validator[0]);
+        }
+
+        bytes32 storedBlsPubKeyHash = checkIndex ? $.validatorIndexes[proof.validatorIndex] : proof.validator[0];
```

Comment:
misleading variable name `storedBlsPubKeyHash` it is not stored if `checkIndex=false` , it gets it from the proof